apps/web/src/components/layout/components/SideNav.tsx

```diff
@@ -45,22 +43,23 @@ type Props = {};
 export function SideNav({}: Props) {
   const navigate = useNavigate();
   const { setEnvironment, isLoading, environment, readonly } = useEnvController();
-  const { currentUser } = useAuthContext();
   const location = useLocation();
   const [opened, setOpened] = useState(readonly);
   const { colorScheme } = useMantineColorScheme();
   const dark = colorScheme === 'dark';
-  const { addItem } = useSpotlightContext();
+  const { addItem, removeItems } = useSpotlightContext();
   const { classes } = usePopoverStyles();
 
   useEffect(() => {
     setOpened(readonly);
     if (readonly && location.pathname === ROUTES.CHANGES) {
       navigate(ROUTES.HOME);
     }
-  }, [readonly]);
+  }, [readonly, navigate, location.pathname]);
```

Comment:
I think we could remove this effect altogether, IMO it is duplicated because we already have navigated to HOME in useEnvController().setEnvironment, which should be triggered every time the env changes similar to `readonly` change

Reply:
You've probably missed that the navigation happens only when the route is `ROUTES.CHANGES`, so like this `useEffect` is guarding that on the prod env no one can navigate to the Changes page. 
But I think that this logic should be in the `PromoteChangesPage` component, so I'll move it there.